src/main/kotlin/core/controller/PokemonController.kt

```diff
@@ -0,0 +1,80 @@
+package core.controller
+
+import core.model.PokemonResponse
+import core.model.PokemonResponseMeta
+import core.model.UserPokemon
+import org.springframework.web.bind.annotation.*
+import core.service.PokemonService
+import core.service.UserService
+
+@RestController
+@RequestMapping("/api/v1/pokemon")
+class PokemonController(val userService: UserService, val pokemonService: PokemonService) {
+    @GetMapping
+    fun getPokemon(
+        @RequestParam(value = "search") search: String?,
+        @RequestParam(value = "perPage", defaultValue = "20") perPage: Int,
+        @RequestParam(value = "page", defaultValue = "1") page: Int,
+        @RequestParam(value = "captured") captured: String?,
+        @RequestHeader("Authorization") authorization: String
+    ): PokemonResponse {
+        val user = userService.validateUser(authorization)
+        var pokemon = pokemonService.findPokemon(user)
```

Comment:
This ends up mutating the `pokemon` variable, which could end up causing some unintended consequences, and makes the code harder to reason about. Instead of getting all the pokemon from the database and then doing the filtering in code, you should make the database do the filtering on `captured`, `search`, and then also the `pagination` as that can all be done in the data layer much more efficiently